src/main/java/org/gelecekbilimde/scienceplatform/auth/service/impl/AuthenticationServiceImpl.java

```diff
@@ -121,9 +124,10 @@ public TokenResponse login(LoginRequest request) {
 		try {
 
 			UserEntity userEntity = userRepository.findByEmail(request.getEmail())
-				.filter(UserEntity::isVerified)
 				.orElseThrow(() -> new UserNotFoundException("User not found: " + request.getEmail()));
-
+			if (!userEntity.isVerified()){
+				throw new VerifyException("user must do the verification process: " + request.getEmail());
```

Comment:
Bu ve diğer hatalarda mesaj constructor'a taşınabilir ve request.getEmail constructor da alınabilir.

Reply:
Bu exception classı bizim yazdığımız değil. Read-only

Reply:
Bu arada classta hiç mesaj görmek istemiyorsanız client exceptionlar için ayrı ayrı exception oluşturabilirim.

Reply:
Spesifik bir hata olması daha sağlıklı olabilir ✅